### ISSUE TITLE
`await-async-events` produces false-positives

### ISSUE THREAD

**Issue by kripod** [testing-library/eslint-plugin-testing-library#881]
### Have you read the Troubleshooting section?

Yes

### Plugin version

v6.2.0

### ESLint version

v8.56.0

### Node.js version

v20.10.0

### package manager and version

pnpm 7.33.7

### Operating system

macOS Sonoma, version 14.3.1

### Bug description

The `userEvent.setup()` method is treated as if it were returning a `Promise` when wrapped by an arrow function expression.

### Steps to reproduce

While exposing a `getUser` wrapper around `userEvent.setup` for reuse:

```ts
export const getUser = () =>
  userEvent.setup({
    advanceTimers: (msToRun: number) => jest.advanceTimersByTime(msToRun),
  });
```

Usages like:

```ts
const user = getUser();
```

Produce a false-positive error as follows:

```
error  Promise returned from `getUser` wrapper over async event method must be handled  testing-library/await-async-events
```

However, `userEvent.setup` isn‘t an async method, so there isn‘t anything to await.

### ESLint configuration

```json
{
  "extends": ["plugin:testing-library/react"]
}
```

or

```json
{
  "plugins": ["testing-library"],
  "rules": {
    "testing-library/await-async-events": "error"
  }
}
```

### Rule(s) affected

`testing-library/await-async-events`

### Do you want to submit a pull request to fix this bug?

No

**Comment by JulianG-TW** [comment]
Just in case it helps.

If I change `getUser` to 
```ts
export const getUser = () => {
  const user = userEvent.setup({
    advanceTimers: (msToRun: number) => jest.advanceTimersByTime(msToRun),
  });
  return user;
}
```
...then the false-positive error is gone.

**Comment by Belco90** [comment]
Thanks for reporting @kripod
